updatejob.go

```diff
@@ -147,7 +152,7 @@ func NewUpdateSettings() *aurora.JobUpdateSettings {
 
 	us := new(aurora.JobUpdateSettings)
 	// Mirrors defaults set by Pystachio
-	us.UpdateOnlyTheseInstances = make(map[*aurora.Range]bool)
+	us.UpdateOnlyTheseInstances = make([]*aurora.Range, 0)
```

Comment:
It shouldn't affect serializing but the problem will come when we try to append. I'd rather take the allocation hit here than have an if nil when adding instances.

Reply:
I don't think there would be problem when append to nil, it will create a new backing array https://play.golang.org/p/DC9c708nGeV 
 https://github.com/golang/go/wiki/CodeReviewComments#declaring-empty-slices

Serialization could be a problem, after all null and empty array can mean different things (optional or no content) and it depends on the protocol format

Reply:
Awesome, since append handles that case then I'll opt for leaving them empty until they are set by an append. The way Thrift will handle this is fine for our needs (if it's there, serialize, if it's not, serialize as empty it not optional or omit if optional).